clang/runtime/dpct-rt/include/dpct/dpl_extras/algorithm.h

```diff
@@ -2414,6 +2414,70 @@ void nontrivial_run_length_encode(ExecutionPolicy &&policy,
   ::std::fill(policy, num_runs, num_runs + 1, ret_dist);
 }
 
+template <typename ExecutionPolicy, typename InputIterator,
+          typename OutputIterator, typename CountIterator,
+          typename UnaryPredicate>
+void partition_if(ExecutionPolicy &&policy, InputIterator input,
+                  OutputIterator output, CountIterator num_true, int num_items,
+                  UnaryPredicate pred, bool reverse_last = true) {
+  dpl::copy(policy, input, input + num_items, output);
+  auto beg_false =
+      dpl::stable_partition(policy, output, output + num_items, pred);
+  auto num_true_items = ::std::distance(output, beg_false);
+  dpl::fill(policy, num_true, num_true + 1, num_true_items);
+  if (reverse_last)
+    dpl::reverse(policy, beg_false, output + num_items);
```

Comment:
Should we be forwarding the policy for this last call? (this is probably impacted by the policy wrapping comment above)

Reply:
Done.